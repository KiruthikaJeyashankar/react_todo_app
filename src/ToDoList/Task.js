import './Task.css';

export default function Task({ task, tellParent }) {

    const toggleTask = () => {
        const newTask = { ...task, completed: !task.completed }
        tellParent(newTask)
    }

    return <>
        {task.completed === true ?
            <li>
                <span onClick={() => toggleTask()}><s>{task.text}</s></span>
            </li>
            :
            <li>
                <span onClick={() => toggleTask()}>{task.text}</span>
            </li>
        }
    </>
};