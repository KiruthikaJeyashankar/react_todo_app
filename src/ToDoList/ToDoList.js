import React from 'react';
import Task from './Task';
import { useState } from "react";

export default function ToDoList({ listOfTasks }) {

  const [todoList, setTodoList] = useState(listOfTasks);

  const handleParent = (updatedTask) => {
    setTodoList((currTodoList) =>
      currTodoList.map((task) =>
        task.id === updatedTask.id ? { ...task, completed: updatedTask.completed } : task
      )
    );
  }

  return (
    <div>
      {todoList.length === 0 && <h4><i>Nothing to do buddy. Sleep!!</i></h4>}
      <ul>
        {todoList.map((task) => <Task key={task.id} task={task} tellParent={(updatedTask) => handleParent(updatedTask)} />)}
      </ul>

      <br /> <br />
      <button data-testid="empty-list-button" type='button' onClick={() => setTodoList([])}>
        Empty List
      </button>

    </div>
  )
}
