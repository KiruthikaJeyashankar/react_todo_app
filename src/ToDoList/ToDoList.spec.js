import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ToDoList from './ToDoList';
import userEvent from "@testing-library/user-event";

describe('ToDoList Tests', () => {
    const task1 = 'Test task 1';
    const task2 = 'Test task 2';
    const listOfTasks = [
        {id: 1, text: task1, completed: false},
        {id: 2, text: task2, completed: false},
    ];

    it('should render todo list and empty list button', () => {
        const component = render(<ToDoList listOfTasks={listOfTasks}/>);

        expect(component.getByRole('list')).toBeInTheDocument();
        expect(component.getByRole('list')).toHaveTextContent('Test task 1Test task 2');
        expect(component.getByTestId('empty-list-button')).toBeInTheDocument();
        expect(component.asFragment()).toMatchSnapshot();
    });

    it('should strike the task when clicked on the task', () => {
        const component = render(<ToDoList listOfTasks={listOfTasks}/>);

        userEvent.click(component.getByText(task1))

        expect(component.getByText(task1)).toHaveStyle('text-decoration: line-through');
    });


    it('should empty list after clicking on "Empty List" button', () => {
        const component = render(<ToDoList listOfTasks={listOfTasks}/>);

        const emptyListButton = component.getByTestId('empty-list-button');
        expect(emptyListButton).not.toBeDisabled();

        userEvent.click(emptyListButton);

        expect(component.getByRole('list')).toHaveTextContent("");
        expect(component.getByText('Nothing to do buddy. Sleep!!')).toBeInTheDocument();
    });
});
