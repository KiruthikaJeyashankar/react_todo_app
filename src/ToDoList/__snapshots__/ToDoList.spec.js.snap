// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList Tests should render todo list and empty list button 1`] = `
<DocumentFragment>
  <div>
    <ul>
      <li>
        <span>
          Test task 1
        </span>
      </li>
      <li>
        <span>
          Test task 2
        </span>
      </li>
    </ul>
    <br />
     
    <br />
    <button
      data-testid="empty-list-button"
      type="button"
    >
      Empty List
    </button>
  </div>
</DocumentFragment>
`;
